components/core/src/clp_s/JsonParser.cpp

```diff
@@ -467,11 +467,19 @@ bool JsonParser::parse() {
                 return false;
             }
 
+            // Add internal log_event_idx field to record
+            auto log_event_idx = add_internal_field(constants::cLogEventIdxName, NodeType::Integer);
```

Comment:
Can we make the order column optional since it may affect the compression ratio a lot for some datasets? And if we want to record  log order, we can call `add_internal_field` only once?

Reply:
If we make it optional I'd like to make it the default + force its use it in the package. I talked to @kirkrodrigues about it and the general consensus we reached is that allowing people to generate archives that don't record order now will cause problems for us down the line.

We could also do some brief experiments to see if we can reduce the overhead (e.g.by trying delta encoding or using 32 bit field size) if that would help convince you. If those kinds of optimizations work we can consider adding them in a later PR.

For `add_internal_field` I'm calling it that we so that we don't need to tie it to every different place where we flush an archive. If you want though I could put it in a lambda and call it in every one of those places.

Reply:
I agree to set it as the default in the package.

For each archive, we can add the internal subtree and the node once, storing the node ID. After that, we just call add_value. Could you clarify what different places you're referring to?

Reply:
Mostly line 506 where `split_archive` is called.

Reply:
Made this change to call `add_metadata_field` once before the ingestion loop and after every invocation of `split_archive`.

Reply:
Also added a `--disable-log-order` option for the compression flow.